app/routes.js

```diff
@@ -25,8 +25,10 @@ import personUtils from './core/personutils';
  *
  * @return {boolean|null} returns true if hash mapping happened
  */
-export const requireAuth = (api) => (nextState, replace) => {
-  if (!api.user.isAuthenticated()) {
+export const requireAuth = (api, store) => (nextState, replace) => {
+  let { blip: state } = store.getState();
+
+  if (!state.isLoggedIn && !api.user.isAuthenticated()) {
```

Comment:
Since `api.user.isAuthenticated` is a sync function, querying the state in this particular `onEnter` hook is not exactly necessary, right? I _think_ I'd prefer to take it out then, for simplicity, but I can probably be persuaded otherwise if I understand your reasoning.